corehq/apps/accounting/tests/test_stripe_payment_method_with_api.py

```diff
@@ -0,0 +1,227 @@
+        self.assertFalse(card["is_autopay"])
+
+    def test_get_card_return_the_correct_card_object(self):
+        # TODO: Should I rename get_card parameter to card_id?
+        card = self.payment_method.get_card(self.card.id)
+        self.assertEqual(card.id, self.card.id)
+
+    def test_get_autopay_card_when_no_autopay_card(self):
+        self.assertEqual(self.billing_account.auto_pay_user, None)
+        self.assertFalse(self.billing_account.auto_pay_enabled)
+        result = self.payment_method.get_autopay_card(self.billing_account)
+        self.assertIsNone(result)
+
+    def test_get_autopay_card_when_only_one_autopay(self):
+        self.payment_method.set_autopay(self.card, self.billing_account, None)
+        result = self.payment_method.get_autopay_card(self.billing_account)
+        self.assertEqual(result.id, self.card.id)
+
+    def test_get_autopay_card_when_one_of_many_card_is_autopay(self):
+        card2 = self.payment_method.create_card('tok_discover', self.billing_account, None, True)
+        card3 = self.payment_method.create_card('tok_amex', self.billing_account, None, True)
+        self.addCleanup(card2.delete)
+        self.addCleanup(card3.delete)
+
+        result = self.payment_method.get_autopay_card(self.billing_account)
+        self.assertIsNotNone(result.id, card2.id)
+
+        result = self.payment_method.get_autopay_card(self.billing_account)
+        self.assertIsNotNone(result.id, card3.id)
+
+    def test_remove_card_successful(self):
+        self.assertEqual(len(self.payment_method.customer.cards), 1)
+        self.payment_method.set_autopay(self.card, self.billing_account, None)
+        self.assertEqual(self.card.id, self.payment_method.get_autopay_card(self.billing_account).id)
+
+        self.payment_method.remove_card(self.card.id)
+        self.assertEqual(len(self.payment_method.customer.cards), 0)
+        self.billing_account.refresh_from_db()
+        self.assertIsNone(self.billing_account.auto_pay_user)
+        self.assertFalse(self.billing_account.auto_pay_enabled)
+
+    def test_remove_card_non_existent(self):
+        with self.assertRaises(stripe.error.InvalidRequestError):
+            self.payment_method.remove_card("non_existent_card_id")
+
+    def test_create_card_creates_card_and_sets_default(self):
+        created_card = self.payment_method.create_card('tok_discover', self.billing_account, None)
+        self.addCleanup(created_card.delete)
+        self.assertIsNotNone(created_card)
+        self.assertEqual(created_card.brand, 'Discover')
+        customer = self.payment_method._get_or_create_stripe_customer()
+        # TODO: not able to get default card updated. But we also didn't use default card at any place.
+        # Can we just remove set default???
+        self.assertEqual(created_card.id, customer.default_card)
+
+    def test_create_charge_success(self):
+        description = "Test charge"
+        amount_in_dollars = 100
```

Comment:
are we sure this isn't a `Decimal`? Isn't the amount that comes from an Invoice always `Decimal`?

Reply:
Thank you!! Fixed all in 1e2e8b4366cc3caadd9f0cacbb8947d85507caa2